pkg/config/migrate/migrate.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2020 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+// Contains all the bits and pieces we need to parse and manage
+// the external configuration
+package migrate
+
+import (
+	"fmt"
+	"github.com/newrelic/infrastructure-agent/pkg/integrations/legacy"
+	"github.com/newrelic/infrastructure-agent/pkg/integrations/v4/config"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+)
+
+const (
+	metricArg      = "metrics"
+	inventoryArg   = "inventory"
+	eventsArg      = "events"
+	exeSuffix      = ".exe"
+	prefixArg      = "--"
+	prefixArgShort = "-"
+)
+
+// V3toV4Result represents the result of a migration
+type V3toV4Result struct {
+	V3toV4Result string `json:"migrateV3toV4Result"`
+}
+
+func V3toV4(pathConfiguration string, pathDefinition string, pathOutput string, overwrite bool) string {
+
+	if _, err := os.Stat(pathOutput); err == nil && !overwrite {
+		log.Fatal(fmt.Errorf("file '%s' already exist and overwrite option is set to false", pathOutput))
+	}
+
+	// Reading old Definition file
+	v3Definition := legacy.Plugin{}
+	err := readAndUnmarshallConfig(pathDefinition, &v3Definition)
+	if err != nil {
+		log.Fatal(fmt.Errorf("error reading old config definition: %w", err))
+	}
+
+	// Reading old Configuration file
+	v3Configuration := legacy.PluginInstanceWrapper{}
+	err = readAndUnmarshallConfig(pathConfiguration, &v3Configuration)
+	if err != nil {
+		log.Fatal(fmt.Errorf("error reading old config configuration: %w", err))
+	}
+
+	// Populating new config
+	v4config, err := populateV4Config(v3Definition, v3Configuration)
+	if err != nil {
+		log.Fatal(fmt.Errorf("error populating new config: %w", err))
+	}
+
+	// Writing output
+	err = writeOutput(v4config, pathDefinition, pathConfiguration, pathOutput)
+	if err != nil {
+		log.Fatal(fmt.Errorf("error writing output: %w", err))
+	}
+
+	return fmt.Sprintf("The config has been migrated and placed in: %s", pathOutput)
```

Comment:
this could return nil (no error) and the caller decide to print it as it has pathOutput too (in case of changing the approach to return error)

Reply:
done